cmd/src/server.rs

```diff
@@ -573,21 +611,8 @@ impl TiKVServer {
 /// If the max open file descriptor limit is not high enough to support
 /// the main database and the raft database.
 fn pre_start(config: &TiKvConfig) {
-    // Sets the global logger ASAP.
-    // It is okay to use the config w/o `validate()`,
-    // because `initial_logger()` handles various conditions.
-    initial_logger(&config);
     tikv_util::set_panic_hook(false, &config.storage.data_dir);
 
-    // Print version information.
-    tikv::log_tikv_info();
-    info!(
-        "using config";
-        "config" => serde_json::to_string(&config).unwrap(),
-    );
-
-    config.write_into_metrics();
-
     check_system_config(&config);
     check_environment_variables();
```

Comment:
This check should be performed before initializing pd client. Because some env var affects gRPC.